app/controllers/incomes_controller.rb

```diff
@@ -0,0 +1,59 @@
+class IncomesController < ApplicationController
+  before_action :require_login
+  before_action :require_current_budget, only: [:show, :edit, :destroy]
+
+  def index
+    @incomes = current_budget.incomes.by_name
+  end
+
+  def new
+    @income = Income.new
+  end
+
+  def create
+    @income = Income.create(income_params.merge(budget: current_budget))
+
+    respond_with(@income)
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def update
+    income.update_attributes(income_params)
+
+    respond_with(income)
+  end
+
+  def destroy
+    income.delete
+
+    respond_with income, location: -> { incomes_path }
+  end
+
+  private
+
+  def income_params
+    params.require(:income).permit(:amount, :due_date, :frequency, :name)
+  end
+
+  def income
+    @income ||= Income.find(params[:id])
```

Comment:
Would be better if we scoped that to be: `current_budget.incomes.find(params[:id])`
Which will raise an `ActiveRecord::NotFound` exception when the income is not found.
And which also means, we don't need the `require_current_budget` before action :)